### ISSUE TITLE
intercept_redirects is failing to load the template due to symfony/web-profiler-bundle depending on @Twig/layout.html.twig 

### ISSUE THREAD

**Issue by ek9** [silexphp/Silex-WebProfiler#109]
The `web_profiler.debug_toolbar.intercept_redirects` feature that was added as part of #34  is currently broken. When intercepting a redirect the profiler tries to render `@WebProfiler/Profiler/toolbar_redirect.html.twig` template which denends on `@Twig/layout.html.twig`.

The problem exception is thrown as we cannot render `@Twig/layout.html` template. There have been comments regarding users experiencing this issue on #34, but as that was closed and not re-opened, I have made a separate / new issue instead.

The interception of redirect happens correct, but when trying to render the page we get this:

![Error while rendering intercept redirect view](http://i.imgur.com/BPhWyT2.png)

```
    in Chain.php line 86
    at Twig_Loader_Chain->getCacheKey('@Twig/layout.html.twig') in Environment.php line 269
    at Twig_Environment->getTemplateClass('@Twig/layout.html.twig') in Environment.php line 345
    at Twig_Environment->loadTemplate('@Twig/layout.html.twig', null) in Template.php line 327
    at Twig_Template->loadTemplate('@Twig/layout.html.twig', '@WebProfiler/Profiler/toolbar_redirect.html.twig', '1') in 1ad0d38760f3b85525a028883166f4446e87877415adee73a7f67be0ff138cf2.php line 11
    at __TwigTemplate_05ac8a8763ed33e05ab0938963b0de02df3f109faeef0ce3ade4571a1f7a2486->__construct(object(Twig_Environment)) in Environment.php line 384
    at Twig_Environment->loadTemplate('@WebProfiler/Profiler/toolbar_redirect.html.twig') in Environment.php line 288
    at Twig_Environment->render('@WebProfiler/Profiler/toolbar_redirect.html.twig', array('location' => '/')) in WebDebugToolbarListener.php line 96
    at WebDebugToolbarListener->onKernelResponse(object(FilterResponseEvent), 'kernel.response', object(TraceableEventDispatcher))
    at call_user_func(array(object(WebDebugToolbarListener), 'onKernelResponse'), object(FilterResponseEvent), 'kernel.response', object(TraceableEventDispatcher)) in WrappedListener.php line 106
    at WrappedListener->__invoke(object(FilterResponseEvent), 'kernel.response', object(EventDispatcher))
    at call_user_func(object(WrappedListener), object(FilterResponseEvent), 'kernel.response', object(EventDispatcher)) in EventDispatcher.php line 174
    at EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.response', object(FilterResponseEvent)) in EventDispatcher.php line 43
    at EventDispatcher->dispatch('kernel.response', object(FilterResponseEvent)) in TraceableEventDispatcher.php line 136
    at TraceableEventDispatcher->dispatch('kernel.response', object(FilterResponseEvent)) in HttpKernel.php line 193
    at HttpKernel->filterResponse(object(RedirectResponse), object(Request), '1') in HttpKernel.php line 175
    at HttpKernel->handleRaw(object(Request), '1') in HttpKernel.php line 68
    at HttpKernel->handle(object(Request), '1', true) in Application.php line 496
    at Application->handle(object(Request)) in Application.php line 477
    at Application->run() in index.php line 10
```
## composer.json

Using latest stable `fabpot/silex-skeleton`. Dependencies:

```json
    "require": {                                                                                                                                                          
        "php": ">=5.5.9",                                                                                                                                                 
        "silex/silex": "~2.0",                                                                                                                                            
        "silex/web-profiler": "^2.0",                                                                                                                                     
        "symfony/asset": "~2.8|3.0.*",                                                                                                                                    
        "symfony/browser-kit": "~2.8|3.0.*",                                                                                                                              
        "symfony/class-loader": "~2.8|3.0.*",                                                                                                                             
        "symfony/config": "~2.8|3.0.*",                                                                                                                                   
        "symfony/console": "~2.8|3.0.*",                                                                                                                                  
        "symfony/css-selector": "~2.8|3.0.*",                                                                                                                             
        "symfony/debug": "~2.8|3.0.*",                                                                                                                                    
        "symfony/finder": "~2.8|3.0.*",                                                                                                                                   
        "symfony/form": "~2.8|3.0.*",                                                                                                                                     
        "symfony/monolog-bridge": "~2.8|3.0.*",                                                                                                                           
        "symfony/process": "~2.8|3.0.*",                                                                                                                                  
        "symfony/security": "~2.8|3.0.*",                                                                                                                                 
        "symfony/translation": "~2.8|3.0.*",                                                                                                                              
        "symfony/twig-bridge": "~2.8|3.0.*",                                                                                                                              
        "symfony/validator": "~2.8|3.0.*"                                                                                                                                 
    },                               
```

Contents of`web/index.php`:

```
use Silex\Application;                                                                                                                                                    
use Silex\Provider\AssetServiceProvider;                                                                                                                                  
use Silex\Provider\TwigServiceProvider;                                                                                                                                   
use Silex\Provider\ServiceControllerServiceProvider;                                                                                                                      
use Silex\Provider\HttpFragmentServiceProvider;                                                                                                                           
use Silex\Provider;                                                                                                                                                       
                                                                                                                                                                          
$app = new Application();                                                                                                                                                 
$app->register(new ServiceControllerServiceProvider());                                                                                                                   
$app->register(new AssetServiceProvider());                                                                                                                               
$app->register(new TwigServiceProvider());                                                                                                                                
$app->register(new HttpFragmentServiceProvider());                                                                                                                        
$app->register(new Provider\HttpFragmentServiceProvider());                                                                                                               
$app->register(new Provider\ServiceControllerServiceProvider());                                                                                                          
$app->register(new Provider\WebProfilerServiceProvider(), array(                                                                                                          
    'profiler.cache_dir' => __DIR__.'/../cache/profiler',                                                                                                                 
    'profiler.mount_prefix' => '/_profiler', // this is the default                                                                                                       
    'web_profiler.debug_toolbar.intercept_redirects' => true, // intercept redirects                                                                                      
));                                                                                                                                                                       
$app['twig'] = $app->extend('twig', function ($twig, $app) {                                                                                                              
    // add custom globals, filters, tags, ...                                                                                                                             
                                                                                                                                                                          
    return $twig;                                                                                                                                                         
});                                                                                                                                                                       
                                                                                                                                                                          
$app->get('/test', function () use ($app) {                                                                                                                               
    return new Symfony\Component\HttpFoundation\RedirectResponse('/');                                                                                                    
});                                                                                                                                                                       
                                                                                                                                                                          
$app['debug'] = true;      
$app->run();                                                                     
```

**Comment by ek9** [comment]
As I am not sure what would be the best way to fix this, I am asking for any input so I could submit a PR for this.

**Comment by hkdobrev** [comment]
I've posted a [$5 bounty on this issue on BountySource](https://www.bountysource.com/issues/42435612-intercept_redirects-is-failing-to-load-the-template-due-to-symfony-web-profiler-bundle-depending-on-twig-layout-html-twig). It goes without saying, but please fix it :) Thank you for all your great work!

**Comment by ek9** [comment]
@hkdobrev I am not sure if this will be accepted by the maintainers (I've tried to request feedback), but I have provided a fix for this on #110 PR.

If you would like to test the fix until this is merged:

1. Add custom repository at the start of `composer.json`:

```json
    "repositories": [                                                                                                                                                     
        {                                                                                                                                                                 
            "type": "vcs",                                                                                                                                                
            "url": "https://github.com/leanphp/Silex-WebProfiler"                                                                                                         
        }                                                                                                                                                                 
    ],               
```

2. Update `silex/web-profilder` to point to `dev-fix-intercept-redirects` branch, like this (in `composer.json`):

```json
    "require": { 
       "silex/web-profiler": "dev-fix-intercept-redirect",
```

You will have to do composer update on `silex/web-profiler` (make sure it installs `dev-fix-intercept-redirect` version). The redirects should work.

Looking for any feedback from the maintainers.

**Comment by stof** [comment]
should be reported to Symfony. The webProfilerBundle should not rely on a layout of TwigBundle for that.
